### PR TITLE
Navitia: Add Brazil providers

### PR DESCRIPTION
**Pull request by grote**
Currently working in São Paulo, Rio de Janeiro, Belo Horizonte and Porto Alegre.

Both providers have been used in Transportr already. Time to upstream.

### CODE REVIEW

File: enabler/src/de/schildbach/pte/AbstractNavitiaProvider.java [hunk 1]
```diff
@@ -721,6 +721,9 @@ public NearbyLocationsResult queryNearbyLocations(final EnumSet<LocationType> ty
 
         try {
             final JSONObject head = new JSONObject(page.toString());
+            if (head.has("error")) {
+                return new NearbyLocationsResult(resultHeader, Status.INVALID_ID);
+            }
```
Comment (grote) [inline enabler/src/de/schildbach/pte/AbstractNavitiaProvider.java]: One of the live tests with an invalid station arrived here and `head` had no `pagination`, but an `error`.

Comment (schildbach) [reply enabler/src/de/schildbach/pte/AbstractNavitiaProvider.java]: Maybe this change should go into a different PR that implements error handling for all Navitia calls?

**Comment by schildbach** [comment]
What's the difference between the two providers. Is one of the datasets likely to be a superset of the other?

**Comment by grote** [comment]
The `BrProvider` is using the official navitia.io server and is currently working in São Paulo, Rio de Janeiro, Belo Horizonte and Porto Alegre.

The `BrFloripaProvider` uses osm2gtfs to create the GTFS which is not available at the official server. I am personally running a server for it. The URL is used in the tests.

**Comment by RockyTV** [comment]
Any update on this?

**Comment by schildbach** [comment]
I just merged the Navitia-based provider.

The other one currently only throws errors. I must admit I'm a bit reluctant on merging – is there any chance this can go to Navitia too? If we continue, this needs a rebase.

**Comment by schildbach** [comment]
Thanks a lot for your contribution. Unfortunately, due to the current bad shape of Navitia support in public-transport enabler, time constraints on our side and due to the centralized nature of navitia.io I have decided to drop support for Navitia. Thus, I'm going to close this PR. Sorry for the inconvenience and thanks again.

**Pull request closed without merging**